packages/sensors_plus/tizen/src/sensors_plus_plugin.cc

```diff
@@ -67,40 +73,129 @@ class SensorsPlusPlugin : public flutter::Plugin {
   virtual ~SensorsPlusPlugin() {}
 
  private:
-  void SetupEventChannels(flutter::PluginRegistrar *registrar) {
+  void SetUpEventChannels(flutter::PluginRegistrar *registrar,
+                          SensorsPlusPlugin *plugin) {
+    std::unique_ptr<FlMethodChannel> method_channel =
+        std::make_unique<FlMethodChannel>(
+            registrar->messenger(), "dev.fluttercommunity.plus/sensors/method",
+            &flutter::StandardMethodCodec::GetInstance());
+
+    method_channel->SetMethodCallHandler(
+        [plugin_pointer = plugin](const auto &call, auto result) {
+          plugin_pointer->HandleMethodCall(call, std::move(result));
+        });
+
     std::unique_ptr<FlEventChannel> accelerometer_channel =
         std::make_unique<FlEventChannel>(
             registrar->messenger(),
             "dev.fluttercommunity.plus/sensors/accelerometer",
             &flutter::StandardMethodCodec::GetInstance());
+    std::unique_ptr<DeviceSensor> accelerometer_sensor =
+        std::make_unique<DeviceSensor>(SensorType::kAccelerometer,
+                                       interval_ms_);
+    accelerometer_sensor_ = accelerometer_sensor.get();
     accelerometer_channel->SetStreamHandler(
         std::make_unique<DeviceSensorStreamHandler>(
-            SensorType::kAccelerometer));
+            std::move(accelerometer_sensor)));
 
     std::unique_ptr<FlEventChannel> gyroscope_channel =
```

Comment:
FYI) [For historical reasons](https://github.com/flutter-tizen/plugins/pull/134), event channel instances are typically owned by plugin classes (see our other plugins that use event channels). However, according to https://github.com/flutter-tizen/plugins/issues/131, this issue has already been resolved and the workaround seems no longer necessary.

That said, it's still good practice to keep the event channel instances alive during the plugin lifecycle. You can update the code accordingly if you think it's better.